core/base/abstractTriangulation/AbstractTriangulation.h

```diff
@@ -2564,11 +2572,51 @@ namespace ttk {
       return this->cellRankArray_;
     }
 
+    inline void setLocalBound(double bound[6]) {
+      this->localBounds_[0] = bound[0];
+      this->localBounds_[1] = bound[1];
+      this->localBounds_[2] = bound[2];
+      this->localBounds_[3] = bound[3];
+      this->localBounds_[4] = bound[4];
+      this->localBounds_[5] = bound[5];
+    };
```

Comment:
I would suggest to use `std::array<double, 6>` instead of C arrays: `std::array`s implement a copy assignment operator, are zero-initialized and don't suffer from array-to-pointer decay.
Besides, they implement a lot of the `std::vector` API (`size()`, iterators) and can be converted to pointers with `data()`.